### ISSUE TITLE
Suggestions/ideas

### ISSUE THREAD

**Issue by Gideon-Foxo** [Gideon-Foxo/support-bot#1]
Currently I plan to make a 'v2' or a rework of this project before I add a lot of new goodies and features. So please note this will take some time, hopefully by the end of summer. 

v2 Changes:

**Better error handling**. 
> Right now error handling is 'okay' but could be a lot better. I would like it to the point where the only errors that are possible are syntax errors (bad config files or moded coded).

**Improved debugging**. 
> Currently dugging is only used for logging commands being loaded... This clearly can be improved.

**More module design**. 
> Most things are in the index.js file and this should be broken up to serval files to process different things and to make it much cleaner.

**Improved setting files**. 
> I am still not fully sure how I want to set this up but with adding a lot new features the settings files need to be restructured.

New Features:

**DM responses**.
> Instead of sending the response in the channel it DMs the user instead. There will be an option of either returning a DMed closed error (telling them to open up their DMs) or an option to just send the message in that channel instead.

**Responses logging**.
> A log of all responses given. E.g. `Gideon requested responses 4 and was DMed successfully`.

**Embed Support**.
> Support for embed responses and logging, ect. I think something like @User [embed with message and color] would look very clean and simple.

**Support Tickets**.
> Instead of granting a role to a user or perms it instead makes a channel for just them and the staff team to view making the support experience much personalized. Optional logs for this would be with txt or maybe md files? Would have to play around with it to make a nice mark down. This would only be logged in a channel and optionally DMed to the user, this wouldn't get stored in a database. 

**Multiple Support System Support**.
In other words allowing there to be more then just 1 support channel system. Not sure how I want to do this yet.  

If you have any more ideas feel free to reply here with them.

**Comment by Gideon-Foxo** [comment]
Look into using dayjs instead of moment https://www.npmjs.com/package/dayjs

**Comment by Gideon-Foxo** [comment]
v2 has been released and most of these features have been added. I will make a project board at some point with all the other ideas I have but at this current time it is not a high priority. If you want something added **please** ask in the Discord server or make an issue to put it on my radar!!!
